apps/console/src/features/users/api/invite.ts

```diff
@@ -0,0 +1,213 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AsgardeoSPAClient, HttpClientInstance } from "@asgardeo/auth-react";
+import { IdentityAppsApiException } from "@wso2is/core/exceptions";
+import { HttpMethods } from "@wso2is/core/models";
+import { AxiosError, AxiosRequestConfig, AxiosResponse } from "axios";
+import { Config } from "../../core/configs";
+import useRequest, { RequestErrorInterface, RequestResultInterface } from "../../core/hooks/use-request";
+import { store } from "../../core/store";
+import { UserManagementConstants } from "../constants";
+import { UserInviteInterface } from "../models";
+
+/**
+ * Initialize an axios Http client.
+ *
+ */
+const httpClient: HttpClientInstance = AsgardeoSPAClient.getInstance().httpRequest.bind(
+    AsgardeoSPAClient.getInstance());
+
+export const getInvitedUserList = (): Promise<any> => {
+    const requestConfig: AxiosRequestConfig = {
+        headers: {
+            "Access-Control-Allow-Origin": store.getState().config.deployment.clientHost,
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: store.getState().config.endpoints.inviteEndpoint
+    };
+
+    return httpClient(requestConfig).then((response: AxiosResponse) => {
+        return Promise.resolve(response);
+    }).catch((error: AxiosError) => {
+        return Promise.reject(error);
+    });
+};
+
+/**
+ * Hook to get the invited users list.
+ * 
+ * @returns invited users list.
+ */
+export const useInvitedUsersList = <Data = UserInviteInterface[],
+    Error = RequestErrorInterface>(
+        shouldFetch: boolean = true
+    ): RequestResultInterface<Data, Error> => {
+
+    const requestConfig: AxiosRequestConfig = {
+        headers: {
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: store.getState().config.endpoints.inviteEndpoint
+    };
+
+    const { data, error, isValidating, mutate } = useRequest<Data, Error>(shouldFetch ? requestConfig : null);
+
+    return {
+        data,
+        error: error,
+        isLoading: !error && !data,
+        isValidating,
+        mutate: mutate
+    };
+};
+
+export const sendInvite = (userInvite: UserInviteInterface): Promise<any> => {
+    const requestConfig: AxiosRequestConfig = {
+        data: userInvite,
+        headers: {
+            "Access-Control-Allow-Origin": store.getState().config.deployment.clientHost,
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.POST,
+        url: store.getState().config.endpoints.inviteEndpoint
+    };
+
+    return httpClient(requestConfig).then((response: AxiosResponse) => {
+        return Promise.resolve(response);
```

Comment:
No need to wrap again in Promise.resolve as httpClient already returns the value as a promise

Reply:
Fixed with fd5839362baf8e4e50397013a20cfa8b0b15fa4e